### PR TITLE
dtspo:242.  Features/dtspo242

### PR DESCRIPTION
**Pull request by mokainos**
New repo.

[ ] Yes
[ x] No
```

### CODE REVIEW

File: .terraform-version [hunk 1]
```diff
@@ -0,0 +1 @@
+0.13.0
```
Comment (timja) [inline .terraform-version:1]:
```suggestion
0.14.3
```

File: azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
+name: HMCTS Shared Services AKS Azure Infrastructure Component Deployment Pipeline
```
Comment (timja) [inline azure-pipelines.yml:1]:
```suggestion
name: SDS Azure Platform
```

File: components/frontdoor/00-init.tf [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  required_version = ">= 0.13.0"
+
+  backend "azurerm" {
+    subscription_id = "04d27a32-7a07-48b3-95b8-3c8691e1a263"
+  }
+  required_providers {
+    azurerm = {
+      source = "hashicorp/azurerm"
+      # https://github.com/terraform-providers/terraform-provider-azurerm/issues/8208
+      version = "2.40.0"
```
Comment (timja) [inline components/frontdoor/00-init.tf:11]:
```suggestion
      version = "2.41.0"
```

File: components/frontdoor/00-init.tf [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    }
+  }
+}
+
+provider "azurerm" {
+  features {}
+  subscription_id = "REPLACE_ME"
```
Comment (timja) [inline components/frontdoor/00-init.tf:18]:
```suggestion
```

File: components/frontdoor/main.tf [hunk 5]
```diff
@@ -0,0 +1,24 @@
+module "landing_zone" {
+  source = "git::https://github.com/hmcts/terraform-module-frontdoor.git?ref=master"
+
+  providers = {
+    azurerm      = azurerm
+    azurerm.data = azurerm.data
```
Comment (timja) [inline components/frontdoor/main.tf:6]: I can't find this actually being used in the module, I don't think it's needed anymore? and should be removed from the module?

File: components/frontdoor/provider.tf [hunk 6]
```diff
@@ -0,0 +1,6 @@
+
+provider "azurerm" {
+  alias = "data"
+  features {}
+  subscription_id = var.data_subscription
+}
```
Comment (timja) [inline components/frontdoor/provider.tf:6]: remove if not required
```suggestion

```

File: pipeline-templates/terraform.yaml [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+  # lowercase $env variable comes from the above template
+
+  - task: Bash@3
+    displayName: 'Subscription Replacement'
+    inputs:
+      targetType: 'inline'
+      workingDirectory: '$(System.DefaultWorkingDirectory)/components/${{parameters.stack}}'
+      script: |
+        sed -i 's|REPLACE_ME|$(ARM_SUBSCRIPTION_ID)|g' '00-init.tf'
```
Comment (timja) [inline pipeline-templates/terraform.yaml:22]: not required
```suggestion
```

File: azure-pipelines.yml [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+      - frontdoor
+      - appgateway
+  - name: location
+    displayName: Location
+    type: string
+    default: 'UK South'
+    values:
+      - 'UK South'
+      - 'UK West'
+  - name: env
```
Comment (timja) [inline azure-pipelines.yml:19]: I would prefer we map to the same convention as in azure-platform-terraform where there's a list of objects that includes all envs and you don't need to run it individually against each env.

with this current approach we don't really know what env something has been run against and it's really easy for environment drift

**Review by timja — COMMENTED** [review]
initial review comments

File: components/appgateway/.terraform-version [hunk 9]
```diff
@@ -0,0 +1 @@
+0.13.4
```
Comment (timja) [inline components/appgateway/.terraform-version:1]:
```suggestion
0.14.3
```

File: components/appgateway/init.tf [hunk 10]
```diff
@@ -0,0 +1,19 @@
+terraform {
+  required_version = ">= 0.13.0"
```
Comment (timja) [inline components/appgateway/init.tf:2]:
```suggestion
  required_version = ">= 0.14.3"
```

File: components/appgateway/init.tf [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+
+  backend "azurerm" {
+    subscription_id = "04d27a32-7a07-48b3-95b8-3c8691e1a263"
+  }
+  required_providers {
+    azurerm = {
+      source = "hashicorp/azurerm"
+      # https://github.com/terraform-providers/terraform-provider-azurerm/issues/8208
+      version = "2.23.0"
```
Comment (timja) [inline components/appgateway/init.tf:11]:
```suggestion
      version = "2.41.0"
```

File: components/frontdoor/init.tf [hunk 12]
```diff
@@ -0,0 +1,17 @@
+terraform {
+  required_version = ">= 0.13.0"
```
Comment (timja) [inline components/frontdoor/init.tf:2]:
```suggestion
  required_version = ">= 0.14.3"
```

File: components/frontdoor/variables.tf [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+
+}
+
+variable "environment" {
+  type = string
+}
+
+
+
+
```
Comment (timja) [inline components/frontdoor/variables.tf:40]:
```suggestion
```

File: pipeline-scripts/builds-check.sh [hunk 14]
```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
```
Comment (timja) [inline pipeline-scripts/builds-check.sh:1]: note: next time we need to add this somewhere we should add it to a pipeline library, non blocking now

File: pipeline-templates/terraform.yaml [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+    name: "CheckFileChanges"
+    displayName: "CheckFileChanges"
+    env:
+      files_list: '$(System.DefaultWorkingDirectory)/components/${{ parameters.stack }} $(System.DefaultWorkingDirectory)/environments/${{ parameters.enviornment }}/${{ parameters.enviornment }}.tfvars'
+    inputs:
+      filePath: pipeline-scripts/file-check-changes.sh
+
+
+
+
```
Comment (timja) [inline pipeline-templates/terraform.yaml:56]:
```suggestion
```

File: components/frontdoor/variables.tf [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+}
+variable "location" {
+  type = string
+}
+variable "frontends" {
+  type = any
+}
+variable "subscription_id" {
+  type = string
+}
```
Comment (timja) [inline components/frontdoor/variables.tf:18]: is this needed?

```suggestion
```

File: environments/demo/demo.tfvars [hunk 17]
```diff
@@ -0,0 +1,19 @@
+project                    = "sdshmcts"
+location                   = "uksouth"
+env                        = "demo"
+subscription               = "demo"
+certificate_key_vault_name = "dtssharedservicesdemokv"
+data_subscription          = "867a878b-cb68-4de5-9741-361ac9e178b6"
+oms_env                    = "demo"
+key_vault_resource_group   = "genesis-rg"
+subscription_id            = "867a878b-cb68-4de5-9741-361ac9e178b6"
```
Comment (timja) [inline environments/demo/demo.tfvars:9]: is this needed?
```suggestion
```

File: components/frontdoor/variables.tf [hunk 18]
```diff
@@ -0,0 +1,40 @@
...
+
+variable "certificate_key_vault_name" {
+  type = string
+}
+variable "oms_env" {
+  type = string
+}
+variable "key_vault_resource_group" {
```
Comment (timja) [inline components/frontdoor/variables.tf:26]: This file should only be stored in one place and then symlinked like in azure-platform-terraform, https://github.com/hmcts/azure-platform-terraform/blob/master/components/cftapps_cluster_lb/variables.tf

that's how we get around the `The root module does not declare a variable named "key_vault_resource_group"` warning there

File: components/frontdoor/main.tf (hunk 5 above)
Comment (mokainos) [reply components/frontdoor/main.tf:14]: terraform frontdoor module needs updating.

File: azure-pipelines.yml (hunk 8 above)
Comment (mokainos) [reply azure-pipelines.yml]: modified parameters in the yaml file

File: components/appgateway/init.tf [hunk 19]
```diff
@@ -0,0 +1,19 @@
+terraform {
+  required_version = ">= 0.14.3"
+
+  backend "azurerm" {
+    subscription_id = "04d27a32-7a07-48b3-95b8-3c8691e1a263"
+  }
+  required_providers {
+    azurerm = {
+      source = "hashicorp/azurerm"
+      # https://github.com/terraform-providers/terraform-provider-azurerm/issues/8208
```
Comment (timja) [inline components/appgateway/init.tf:10]:
```suggestion
```

File: components/appgateway/init.tf [hunk 20]
```diff
@@ -0,0 +1,19 @@
+terraform {
+  required_version = ">= 0.14.3"
```
Comment (timja) [inline components/appgateway/init.tf:2]:
```suggestion
  required_version = ">= 0.14.4"
```

File: components/frontdoor/init.tf [hunk 21]
```diff
@@ -0,0 +1,23 @@
+terraform {
+  required_version = ">= 0.14.3"
```
Comment (timja) [inline components/frontdoor/init.tf:2]:
```suggestion
  required_version = ">= 0.14.4"
```

**Review by timja — APPROVED** [review]
couple of minor things, could you fix them up before merge, but :+1: from me

**Pull request merged**